src/main/java/bio/overture/ego/security/CustomOAuth2UserInfoService.java

```diff
@@ -0,0 +1,132 @@
+package bio.overture.ego.security;
+
+import static java.util.Objects.isNull;
+
+import bio.overture.ego.model.enums.ProviderType;
+import bio.overture.ego.service.GithubService;
+import bio.overture.ego.service.LinkedinService;
+import bio.overture.ego.service.OrcidService;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.val;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.oauth2.client.userinfo.DefaultOAuth2UserService;
+import org.springframework.security.oauth2.client.userinfo.OAuth2UserRequest;
+import org.springframework.security.oauth2.core.OAuth2AuthenticationException;
+import org.springframework.security.oauth2.core.user.DefaultOAuth2User;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+@Component
+public class CustomOAuth2UserInfoService extends DefaultOAuth2UserService {
```

Comment:
This service is needed to fetch any extra data that we need about the user after we receive the access token

Reply:
this is for oauth2 providers that are not openid connect